Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -2122,7 +2122,13 @@ static NSTrackingAreaOptions flagsChangedEventMonitorTrackingAreaOptions()
 void WebViewImpl::windowWillBeginSheet()
 {
 #if ENABLE(POINTER_LOCK)
-    m_page->requestPointerUnlock();
+    m_page->requestPointerUnlock([this](bool result) {
+        if (result) {
+            WeakPtr webContentPointerLockProcess = m_page->webContentPointerLockProcess();
```

Comment:
This needs to be a RefPtr.
Also, instead of webContentPointerLockProcess and clearWebContentPointerLockProcess, why not just make one function takePointerLockProcess?